### PR TITLE
case indent and space after cast  check

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**

### CODE REVIEW

File: ngx-style.pl [hunk 1]
```diff
@@ -144,7 +144,7 @@
             }
 
             # 3.
-            if ($line =~ /\(\w+\)\w+/) {
+            if ($line =~ /\([a-zA-Z0-9_*]+( \*)?\)[a-zA-Z_]+/) {
```
Comment (agentzh) [inline ngx-style.pl:147]: This looks wrong. We should give warning/error messages for things like `(foo*)` or `(foo    *)`.

Comment (zhuizhuhaomeng) [reply ngx-style.pl]: this is check in other place. we check （type)p  or (type *)p here.

File: ngx-style.pl [hunk 2]
```diff
@@ -175,6 +175,11 @@
                     output "incorrect front spaces, unclosed bracket";
                 }
 
+                # skip fall through case
+                if ($line =~ /^ +(?:case).*$/) {
```
[the remark this answers is not in the document]
Comment (zhuizhuhaomeng) [reply ngx-style.pl]: wo should have a space after case so i added a space after it

File: ngx-style.pl [hunk 3]
```diff
@@ -175,6 +175,11 @@
                     output "incorrect front spaces, unclosed bracket";
                 }
 
+                # skip fall through case
+                if ($line =~ /^ +(?:case ).*:/) {
```
Comment (agentzh) [inline ngx-style.pl:179]: Why do you need `(?:  )` here in the first place?

**Comment by agentzh** [comment]
@zhuizhuhaomeng Will you try the following extra patch on your side? I fixed some minor things and also added one extra check:

```diff
diff --git a/ngx-style.pl b/ngx-style.pl
index 118ff40..634949f 100755
--- a/ngx-style.pl
+++ b/ngx-style.pl
@@ -176,7 +176,11 @@ for my $file (@ARGV) {
                 }

                 # skip fall through case
-                if ($line =~ /^ +(?:case ).*:/) {
+                if ($line =~ /^\s*case\b(\s*).*:/) {
+                    if ($1 ne ' ') {
+                        output "incorrect use of whitespace chars after 'case' "
+                               . "(one and only one space is required here)";
+                    }
                     $next_level = 0;
                 }

@@ -199,7 +203,8 @@ for my $file (@ARGV) {

             # enter next level state
             if ($macro_defined == 0
-                && ($line =~ /^((?<!switch).)*{\n$/ || $line =~ /^ +case .*:/)) {
+                && ($line =~ /^((?<!switch).)*{\n$/ || $line =~ /^\s*case\b.*?:/))
+            {
                 $next_level = 1;
                 $next_level_space = $space + 4;
             }
```

**Comment by agentzh** [comment]
Merged. Thanks.

**Pull request closed without merging**